core/app/models/spree/shipment.rb

```diff
@@ -200,21 +200,25 @@ def selected_shipping_rate_id=(id)
       end
     end
 
-    # Determines the appropriate +state+ according to the following logic:
+    # Assigns the appropriate +state+ according to the following logic:
     #
     # canceled   if order is canceled
     # pending    unless order is complete and +order.payment_state+ is +paid+
     # shipped    if already shipped (ie. does not change the state)
     # ready      all other cases
-    def determine_state(order)
-      return 'canceled' if order.canceled?
-      return 'shipped' if shipped?
-      return 'pending' unless order.can_ship?
-      if can_transition_from_pending_to_ready?
-        'ready'
-      else
-        'pending'
-      end
+    def recalculate_state
```

Comment:
I'm not sure I fully understand the benefits of assigning the value to the instance of the object, if we are using this method only when we cannot use the instance as is (just calling `save`). It might look cleaner, but I have the feeling we are just moving the complexity into `update_state`. Let me know your thoughts (non-blocking comment).

Reply:
The benefit of assigning the value is a bit abstract at this point, but it's in service of the in-memory order updater (#5872). Using an assignment is great in situations where we don't actually want to persist changes to an order but want to ready or preview for a future state. Currently, we would have to use `#determine_state` to do this, but that doesn't let us actually re-use that value in memory for the current shipment.

> It might look cleaner, but I have the feeling we are just moving the complexity into `update_state`.

I am not sure I understand what you mean. I don't think the functionality of `#update_state` will be changing at all as a result of this.